src/coreclr/jit/gentree.cpp

```diff
@@ -3964,9 +3966,32 @@ unsigned Compiler::gtSetEvalOrder(GenTree* tree)
                             goto DONE;
                         }
                     }
+#endif
+
+                    switch (hwTree->gtHWIntrinsicId)
```

Comment:
Do we need to review costing for HW intrinsics more broadly?

Reply:
We likely should. We probably aren't accounting for cases where helper intrinsics are more expensive than they appear nor cases where operands have less cost due to special handling that hardware intrinsics get.

There's also probably cases where operands (like scalar DBL_CNS) are currently participating in overall CSE and shouldn't for certain cases.